src/jit/gentree.cpp

```diff
@@ -7971,51 +7984,118 @@ GenTreePtr Compiler::gtReplaceTree(GenTreePtr stmt, GenTreePtr tree, GenTreePtr
             treeLastNode->gtNext = treeNextNode;
             treeNextNode->gtPrev = treeLastNode;
         }
-
-        // Propagate side-effect flags of "replacementTree" to its parents if needed.
-        gtUpdateSideEffects(treeParent, tree->gtFlags, replacementTree->gtFlags);
     }
 
     return replacementTree;
 }
 
 //------------------------------------------------------------------------
-// gtUpdateSideEffects: Update the side effects for ancestors.
+// gtUpdateSideEffects: Update the side effects for statement tree nodes.
+//
+// Arguments:
+//    stmt            - The statement to update side effects on
```

Comment:
You need a description of the `tree` argument. Also, since `tree` is unused in the then clause below, and `stmt` is unused in the else clause, would it make sense to have two different versions of this method, especially since their behavior is quite different - in one case an entire statement is updated, and in another case, only the direct children of `tree`, and its ancestors.